js/app.js

```diff
@@ -6,7 +6,8 @@
 var tagModel = new TagModel();
 var textArea = $('#doc-view');
 var highlightArea = $('#highlightArea');
-var label_list = $("#label-list");
+var label_list = $('#label-list');
+var label_selected = $('#label-selected');
```

Comment:
And then for label deletion/updates in the left panel, i think this is the problem.  Killing the label_selected var and just using the `$('#label-selected');` seems to fix the update problems.  At a guess, i'd say maybe this only updates when app.js loads, but doesn't update whenever the var label_selected is referenced, so it's holding some old data, but who knows.  maybe worth it just to keep 'em in the jQuery stuff